agent/agent_test.go

```diff
@@ -256,12 +256,13 @@ func TestAgent_Stats_Magic(t *testing.T) {
 		_ = tunneledConn.Close()
 
 		require.Eventuallyf(t, func() bool {
-			var ok bool
-			s, ok = <-stats
-			return ok && s.ConnectionCount == 0 &&
+			s, ok := <-stats
+			t.Logf("got stats after disconnect %t, %d",
+				ok, s.SessionCountJetBrains)
+			return ok &&
```

Comment:
TSMP ping counts as a "connection" in the agent stats so it never goes to zero.  I'm not sure why we were testing it going to zero in the first place @code-asher

Reply:
I think we wanted to make sure `ConnectionCount` was also updating correctly, but did not realize there would be other connections going on.